src/cloud/audiocom/ShareAudioDialog.cpp

```diff
@@ -0,0 +1,842 @@
+
+      wxFileName fileName(
+         pathName + "/cloud/",
+         wxString::Format("%lld", std::chrono::system_clock::now().time_since_epoch().count()),
+         GetExtension());
+
+      fileName.Mkdir(0700, wxPATH_MKDIR_FULL);
+
+      if (fileName.Exists())
+      {
+         if (!wxRemoveFile(mPath))
+            return false;
+      }
+
+      mPath = fileName.GetFullPath();
+
+      return true;
+   }
+
+   bool Export(std::unique_ptr<BasicUI::ProgressDialog>& progress)
+   {
+      if (!GenerateTempPath())
+         return false;
+
+      SettingScope scope;
+
+      SetupFormat();
+
+      Exporter e { const_cast<AudacityProject&>(mProject) };
+
+      auto& tracks = TrackList::Get(mProject);
+
+      const double t0 = 0.0;
+      const double t1 = tracks.GetEndTime();
+
+      const int nChannels = (tracks.Any() - &Track::IsLeader).empty() ? 1 : 2;
+
+      return e.Process(
+         nChannels,   // numChannels,
+         GetFormat(), // type,
+         mPath,       // full path,
+         false,       // selectedOnly,
+         t0,          // t0
+         t1,          // t1
+         progress     // progress dialog
+      );
+   }
+};
+
+struct WavExportHelper final : ExportHelper
+{
+   explicit WavExportHelper(AudacityProject& project)
+       : ExportHelper(project)
+   {
+   }
+
+   void SetupFormat() override
+   {
+      // We have very little control over the WAV export:
+      // all the settings are stored based on SND file constants
+   }
+
+   FileExtension GetFormat() const override
+   {
+      return FileExtension { "WAV" };
+   }
+
+   FileExtension GetExtension() const override
+   {
+      return FileExtension { "wav" };
+   }
+};
+
+#ifdef USE_LIBFLAC
+struct FlacExportHelper final : ExportHelper
+{
+   explicit FlacExportHelper(AudacityProject& project)
+       : ExportHelper(project)
+   {
+   }
+
+   void SetupFormat() override
+   {
+      FLACBitDepth.Write("24");
+      FLACLevel.Write("5");
+   }
+
+   FileExtension GetFormat() const override
+   {
+      return FileExtension { "FLAC" };
+   }
+
+   FileExtension GetExtension() const override
+   {
+      return FileExtension { "flac" };
+   }
+};
+#endif
+
+#ifdef USE_WAVPACK
+struct WavPackExportHelper final : ExportHelper
+{
+   explicit WavPackExportHelper(AudacityProject& project)
+       : ExportHelper(project)
+   {
+   }
+
+   void SetupFormat() override
+   {
+      QualitySetting.Write(2);
+      BitrateSetting.Write(40);
+      BitDepthSetting.Write(24);
+      HybridModeSetting.Write(false);
+   }
+
+   FileExtension GetFormat() const override
+   {
+      return FileExtension { "WavPack" };
+   }
+
+   FileExtension GetExtension() const override
+   {
+      return FileExtension { "wv" };
+   }
+};
+#endif
+
+// We use WAV only as a fall back (unless config asks for it explicitly)
+std::unique_ptr<ExportHelper>
+GetExportHelper(AudacityProject& project)
+{
+   const auto preferredFormat = GetServiceConfig().GetPreferredAudioFormat();
```

Comment:
Here it is... that preference that I noted before, is not a preference.  It's always WAV.  No memory of a user choice.  Must that be fixed for release?

Reply:
I should be always wavpack (if available), I will tipple check that. There are no plans to give the user control of exporting right now, so this is not a bug.

Reply:
And it is part of service *configuration*, it is not a preference.

Reply:
It's still true in the latest push that the return value of this function never changes.  Is it your intent to check availability of WavPack?

I think I know how to do that.  Wait for me to push a PR onto your PR.